array-reduce/main.js

```diff
@@ -0,0 +1,45 @@
+const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12];
+
+const account = [
+  { type: 'deposit', amount: 150 },
+  { type: 'deposit', amount: 20 },
+  { type: 'withdrawal', amount: 5 },
+  { type: 'deposit', amount: 100 },
+  { type: 'withdrawal', amount: 25 },
+  { type: 'withdrawal', amount: 60 }
+];
+
+const traits = [
+  { color: 'yellow' },
+  { type: 'electric' },
+  { name: 'pikachu' },
+  { level: 15 },
+  { trainer: 'ash' }
+];
+
+const reduce1 = (accumulator, currentValue) => accumulator + currentValue;
+const sum = numbers.reduce(reduce1);
+console.log("Sum: ", sum);
+
+const reduce2 = (accumulator, currentValue) => accumulator * currentValue;
+const product = numbers.reduce(reduce2);
+console.log("Product: ", product);
+
+const depositArray = [];
+const withdrawalArray = [];
+for(let i = 0; i < account.length; i++){
+  if(account[i].type === 'deposit'){
+    depositArray.push(account[i].amount);
+  }
+  if(account[i].type === 'withdrawal'){
+    withdrawalArray.push(account[i].amount);
+  }
+}
+const reduce3 = (accumulator, currentValue) => accumulator + currentValue;
+const totalDeposit = depositArray.reduce(reduce3);
+const totalWithdrawal = withdrawalArray.reduce(reduce3);
+console.log("Balance: ", totalDeposit - totalWithdrawal);
```

Comment:
This should be done using the `.reduce` method. You don't need to use a separate `for` loop.

```JAVASCRIPT
const balance = account.reduce((total, transaction) => {
  // If the transaction is a deposit add the "amount" to the "total"
  
  // if the transaction is a withdrawal subtract the "amount" from the "total"
  
  // return the "total"
}, 0)
```